napari/layers/_data_protocols.py

```diff
@@ -33,10 +35,15 @@ def _raise_protocol_error(obj: Any, protocol: type):
     annotations = getattr(protocol, '__annotations__', {})
     needed = set(dir(protocol)).union(annotations) - _OBJ_NAMES
     missing = needed - set(dir(obj))
-    message = (
-        f"Object of type {type(obj).__name__!r} does not implement "
-        f"{protocol.__name__!r} Protocol.\n"
-        f"Missing methods: {missing!r}"
+    type_name = repr(type(obj).__name__)
+    protocol_name = repr(protocol.__name__)
+    missing_methods = repr(missing)
+    message = trans._(
```

Comment:
This is one example where I'm unsure if this is desired, and if this is the best approach. Would be fine reverting this for now and addressing later.

Reply:
I don't know either, but that looks fine to me.